config/src/main/java/com/typesafe/config/ConfigFactory.java

```diff
@@ -1090,6 +1090,63 @@ public static Config parseResourcesAnySyntax(String resourceBasename) {
         return parseResourcesAnySyntax(resourceBasename, ConfigParseOptions.defaults());
     }
 
+    /**
+     * Parse only any application overrides (those specified by config.{resource,file,url}), returning
+     * an empty Config if no overrides were set.
+     * @param parseOptions parse options
+     * @return the parsed configuration
+     */
+    public static Config parseApplicationOverride(ConfigParseOptions parseOptions) {
+        ClassLoader loader = parseOptions.getClassLoader();
+
+        if (loader == null)
+            throw new ConfigException.BugOrBroken(
+                "ClassLoader should have been set here; bug in ConfigFactory. "
+                    + "(You can probably work around this bug by passing in a class loader or calling currentThread().setContextClassLoader() though.)");
```

Comment:
this should probably follow the pattern in the rest of ConfigFactory and call `ensureClassLoader` rather than expecting it to be already set. The assertion here was because ConfigFactory is supposed to be the only caller of the strategy and it always does the ensureClassLoader. So here you can use ensureClassLoader which already throws if it can't find a loader.